src/liballoc/boxed.rs

```diff
@@ -596,6 +596,40 @@ impl From<Box<str>> for Box<[u8]> {
     }
 }
 
+use core::convert::TryFrom;
+
+#[derive(Debug, Copy, Clone)]
+#[unstable(feature = "boxed_slice_try_from", issue = "0")]
+/// Can we use TryFromSliceError instead?
+/// TODO: implement Error
+pub struct PlaceholderError(());
+
+macro_rules! array_impls {
+    ($($N:expr)+) => {
+        $(
+            #[unstable(feature = "boxed_slice_try_from", issue = "0")]
+            impl<T> TryFrom<Box<[T]>> for Box<[T; $N]> {
+                type Error = PlaceholderError;
+
+                fn try_from(boxed_slice: Box<[T]>) -> Result<Box<[T; $N]>, Self::Error> {
+                    if boxed_slice.len() == $N {
+                        unsafe { Ok(Box::from_raw(Box::into_raw(boxed_slice) as *mut [T; $N])) }
```

Comment:
A comment re. safety would be nice.

Reply:
I wanted to copy the rationale from slices, but it didn't have one; neither does the `unsafe` block immediately before this implementation in the file.

Is it valid to say

> This is safe for the same reasons and under the same conditions that converting a slice to an array is safe

?

Reply:
I would elaborate more from the POV of the unsafe operations you are using and how you have verified the proofs they need. Also, `Ok(unsafe {` seems preferable since the `Ok` bit doesn't partake in the unsafety.